include/knowhere/expected.h

```diff
@@ -83,6 +83,16 @@ class expected {
         return val.value();
     }
 
+    const std::string&
+    what() const {
+        return msg;
+    }
```

Comment:
I feel it would be better if the message is attached to a non success `Status` instead of an `excepted`: we may have some error info to be returned even if the current method returns only `Status`.

But now our Status is an enum class and cannot be extended. Would be a large change if we want to do that.

See [absl::Status](https://abseil.io/docs/cpp/guides/status) for reference.

Reply:
In your proposed change to pass err_msg to the returned value of Search the code looks like this, which is pretty clumsy:

```cpp

expected<DataSetPtr> Search() const {
    auto cfg = this->node->CreateConfig();
    std::string err;
    auto status = LoadConfig(cfg.get(), json, knowhere::SEARCH, "Search", &err);
    if (err != Status::success) {
        expected<DataSetPtr> ret(err);
        ret << err;
        return ret;
    }
    ...
}

Status
LoadConfig(BaseConfig* cfg, const Json& json, knowhere::PARAM_TYPE param_type, const std::string& method, std::string* const err = nullptr) const {
    Json json_(json);
    auto res = Config::FormatAndCheck(*cfg, json_, err);
    LOG_KNOWHERE_DEBUG_ << method << " config dump: " << json_.dump();
    RETURN_IF_ERROR(res);
    return Config::Load(*cfg, json_, param_type, err);
}
```

Reply:
change the return status, Leads to huge changes.
err_msg, Compatible with old code.
I think just add err_msg for search api.